code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -354,262 +363,265 @@
+	data["ai_tracking"] = untrackable
+	return data
+
+/obj/item/card/id/syndicate/ui_static_data(mob/user)
+	var/list/data = list()
+	var/list/idcard_skins = list()
+	for(var/idcard_skin in get_all_card_skins())
+		idcard_skins.Add(list(list(
+			"name" = idcard_skin,
+		)))
+	data["idcards"] = idcard_skins
+	return data
+
+
+/obj/item/card/id/syndicate/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/idcards),
+		get_asset_datum(/datum/asset/spritesheet/orbit_job),
+	)
+
+/obj/item/card/id/syndicate/ui_state(mob/user)
+	return GLOB.default_state
+
+/obj/item/card/id/syndicate/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "AgentCard", name)
+		ui.open()
+
+/obj/item/card/id/syndicate/attack_self(mob/user)
+	if(!ishuman(user))
+		return
+	if(!registered_human)
+		registered_human = user
+	if(registered_human != user)
+		flash_card(user)
+		return
+	switch(tgui_alert(user, "Would you like to display [src] or edit it?", "Choose", list("Edit", "Show")))
+		if("Show")
+			flash_card(user)
+		if("Edit")
+			ui_interact(user)
+
+/obj/item/card/id/syndicate/proc/delete_info(datum/tgui/ui)
+	var/response = tgui_alert(registered_human, "Are you sure you want to delete all card info?","Delete Card Info", list("Yes", "No"))
+	if(!Adjacent(registered_human) || isnull(response))
+		return
+	if(response == "Yes")
+		name = initial(name)
+		registered_name = initial(registered_name)
+		icon_state = initial(icon_state)
+		sex = initial(sex)
+		age = initial(age)
+		assignment = initial(assignment)
+		rank = initial(rank)
+		associated_account_number = initial(associated_account_number)
+		blood_type = initial(blood_type)
+		dna_hash = initial(dna_hash)
+		fingerprint_hash = initial(fingerprint_hash)
+		photo = null
+		registered_human.sec_hud_set_ID()
+		registered_human = null
+		ui.close()
+
+/obj/item/card/id/syndicate/proc/clear_access()
+	var/response = tgui_alert(registered_human, "Are you sure you want to reset access saved on the card?","Reset Access", list("Yes", "No"))
+	if(!Adjacent(registered_human) || isnull(response))
+		return
+	if(response == "Yes")
+		access = initial_access.Copy() // Initial() doesn't work on lists
+		to_chat(registered_human, "<span class='notice'>Card access reset.</span>")
+
+/obj/item/card/id/syndicate/proc/change_ai_tracking()
+	untrackable = !untrackable
+
+/obj/item/card/id/syndicate/proc/change_name()
+	var/new_name = reject_bad_name(tgui_input_text(registered_human,"What name would you like to put on this card?","Agent Card Name", ishuman(registered_human) ? registered_human.real_name : registered_human.name), TRUE)
+	if(!Adjacent(registered_human) || isnull(new_name))
+		return
+	registered_name = new_name
+	UpdateName()
+
+/obj/item/card/id/syndicate/proc/change_photo()
+	var/job_clothes = null
+	if(assignment)
+		job_clothes = assignment
+	var/icon/newphoto = get_id_photo(registered_human, job_clothes)
```

Comment:
This can be an expensive proc, we should probably put a cooldown on regenerating this photo, you can find the helpers for cooldowns in tg_cooldowns.dm